### ISSUE TITLE
'Unexpected an empty line before property "undefined"'

### ISSUE THREAD

**Issue by jamesarosen** [hudochenkov/stylelint-order#87]
Given this stylelint configuration:
```js
'order/properties-order': [
  [
    'color',
    'font-size',
  ],
  {
    unspecified: 'bottom',
    emptyLineBeforeUnspecified: 'never',
  },
]
```
and the following SCSS:
```scss
.foo {
  color: blue;
  font-size: 12px;

  &__bar {
    color: purple;
  }
}
```

I get the following error:
```
   5:3  ✖  Unexpected an empty line before property "undefined"            order/properties-order
```

**Comment by hudochenkov** [comment]
Thank you for the report! I'll take a look in a few days.

**Comment by hudochenkov** [comment]
Indeed. Bug is happening when `emptyLineBeforeUnspecified` is specified.

**Comment by hudochenkov** [comment]
I've fixed some existing incorrect tests for `emptyLineBeforeUnspecified` and added more https://github.com/hudochenkov/stylelint-order/pull/90. Now we need to fix the code.

@njbraun would you like to help fix this option, please?

**Comment by njbraun** [comment]
@jamesarosen I want to make sure we're capturing your issue correctly.  From the example you gave, I'm potentially seeing 3 issues:

1) The error message incorrectly identifies the ```&__bar {
    color: purple;
  }``` block as `"undefined"`  Instead, the error message should read: `5:3  ✖  Unexpected an empty line before property "&__bar {"            order/properties-order`

2) The error message `5:3  ✖  Unexpected an empty line before property "undefined"            order/properties-order` shouldn't include the word `an` between Unexpected and empty.

3) The error itself is unexpected.  You don't count the line `&__bar {` as an "unspecified" item, so this rule should not be active and the example you gave should pass stylelint without issue.

Let us know why this was originally filed.

**Comment by hudochenkov** [comment]
@njbraun there should not be any violations, because there are no unspecified properties.

**Comment by jamesarosen** [comment]
I am specifically referring to (3)

**Comment by hudochenkov** [comment]
Fix released in `stylelint-order@4.0.0`. Happy New Year!
